components/logins/src/db.rs

```diff
@@ -555,126 +536,131 @@ impl LoginDb {
         Ok(metrics)
     }
 
-    pub fn update(&self, login: Login) -> Result<()> {
-        let login = self.fixup_and_check_for_dupes(login)?;
-
+    pub fn update(
+        &self,
+        sguid: &str,
+        login: UpdatableLogin,
+        encdec: &EncryptorDecryptor,
+    ) -> Result<Login> {
+        let guid = Guid::new(sguid);
+        let now_ms = util::system_time_ms_i64(SystemTime::now());
         let tx = self.unchecked_transaction()?;
+
+        // XXX - it's not clear that throwing here on a dupe is the correct thing to do - eg, a
+        // user updated the username to one that already exists - the better thing to do is
+        // probably just remove the dupe.
```

Comment:
I agree about removing the dupe. The only caveat I'd have is considering what would happen if the dupe has an older `time_password_changed` timestamp. In that case we probably want to keep the dupe or at minimum the dupe's metadata.

Reply:
I think an important consideration here will be the existing Fenix code - eg, will the UI update correctly if we do that deletion? It would clearly be bad if the login we removed stayed in the list of logins. Sadly I can't build Fenix currently for reasons I don't understand, so can't check.

The metadata is an interesting question - maybe we should delete the one being updated as that has entirely wrong metadata now that one of the keys has changed.